qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1134,36 +1135,69 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 	 * @param projectUri  the project Uri.
 	 * @param diagnostics the diagnostic list.
 	 */
-	private void validateCaseSectionParameters(Section caseSection, Section whenSection, ResolvedJavaTypeInfo whenType,
-			String projectUri, List<Diagnostic> diagnostics) {
+	private void validateCaseSectionParameters(CaseSection caseSection, Section whenSection,
+			ResolvedJavaTypeInfo whenType, String projectUri, List<Diagnostic> diagnostics) {
 		String expression = whenSection.getExpressionContent();
 		List<Parameter> parameters = caseSection.getParameters();
 		String whenTypeName = whenType.getSignature();
+		Parameter firstParameter = parameters.get(0);
 		// Iterate through children and ensure the type of the #case or #is parts are
 		// the same as the parent section
 		for (int i = 0; i < parameters.size(); i++) {
 			Parameter parameter = parameters.get(i);
 			String parameterName = parameter.getName();
-			if (i == 0 && parameters.size() > 1) {
-				// The first is allowed as an operator so it is skipped for, second must be
-				// checked
-				// Ex: {#is in BROKEN}
+			Range range = QutePositionUtility.createRange(parameter);
+			if (i == 0 && caseSection.isCaseOperator(firstParameter)) {

```

Comment:
You call several times `caseSection.isCaseOperator(firstParameter)` that it seach several time the map of operators. I suggest that you create a CaseOperator variable 

`CaseOperator operator = caseSection.getCaseOperator(firstParameter) `

and after that you write

```
if (caseOperator == null) {
   // Validation when the first parameter is not an operator
    if (i > 0) {
        ...
    }
} else {
   // Validation when the first parameter is an operator
    if (i == 0) {
        continue;
    } else if (i > 1 && !caseOperator.isMulti()) {

    }

}
```